catalyst/dl/scripts/run.py

```diff
@@ -116,14 +116,14 @@ def main_worker(args, unknown_args):
     config.setdefault("distributed_params", {})["apex"] = args.apex
     config.setdefault("distributed_params", {})["amp"] = args.amp
 
+    if args.logdir is not None and get_rank() <= 0:
+        dump_environment(config, args.logdir, args.configs)
+        dump_code(args.expdir, args.logdir)
```

Comment:
the tricky problem here:
- Catalyst supports the` baseligdir` option to drop your experiment in one place with unique name support
https://github.com/catalyst-team/catalyst/blob/master/catalyst/experiments/config.py#L106#L118

in this way, you have to fix expreriment_params in the `prepare_config_api_components` function :)

Reply:
Will do:)

Reply:
But the idea of preparing the config first and then pass it as an argument to experiment and runner initialization I like the most🤔

Reply:
you mean, to create an Experiment, and then re-define `logdir` for the second Experiment?

Reply:
o, I mean the following:
move `prepare_logdir` and `get_logdir` from ConfigExperiment https://github.com/catalyst-team/catalyst/blob/0a2afda0e90425c206f3ece3a28e347cb58278d3/catalyst/experiments/config.py#L94-L112
to some different location (maybe in utils) and get _logdirs_ (set or generated if _baselogdir_ is set) in `main_worker` before resetting environment and even before initializing experiment and runner.

Or should it be strictly generated in ConfigExperiment?..

Reply:
🤔  here is the trick: we would like to give the user a convenient way to specify logdir based on `baselogdir`
and that's why we want to keep `prepare_logdir` inside Experiment

Reply:
But now `prepare_config` gust uses config and _baselogdir_ to generate a new _logdir_ as a hash of time and input config.
Do you want to add other features in the future?

Reply:
It only defaults :)
you can easily rewrite it with
```python

from catalyst import registry

registry.Experiment
class SomeExperimnet:
  def prepare_logdir(...):
    return self. baselogdir  + (self.config["num_1"] _ self.config["num_2"] _ etc)
```

In this way, you could easily save your experiments with some predefined hparams-naming

Reply:
Got it!
Ok then, then I will change `prepare_config_api_components`:)
But not as in the last time, because the name of the custom runner also disappears. The idea - to make a deepcopy of config at the beginning of the function which is used for experiment and runner, and return the initial config in the end.

Reply:
sounds legit, let's make it 🚀